virttest/qemu_vm.py

```diff
@@ -2665,15 +2665,21 @@ def create(self, name=None, params=None, root_dir=None,
                             net_mask=params.get("net_mask", None),
                             pa_type=pa_type)
 
+                    if nic_params.get("device_name").split(':', 1)[0] == "shell":
```

Comment:
if  nic_params.get("device_name", "").startswith("shell:"):

Reply:
I have updated and tests are pass.